ods_ci/tests/Tests/0100__platform/0101__deploy/0102__auth_providers_and_rbac/0103__auth_crd.robot

```diff
@@ -0,0 +1,56 @@
+*** Settings ***
+Documentation       Test Suite for Auth CRD functionality verification
+
+Resource            ../../../../Resources/ODS.robot
+Resource            ../../../../Resources/Page/OCPDashboard/OCPDashboard.resource
+Resource            ../../../../Resources/RHOSi.resource
+
+Suite Setup         Auth Crd Suite Setup
+Suite Teardown      Auth Crd Suite Teardown
+
+
+*** Test Cases ***
+Verify Auth Crd
+    [Documentation]    Happy path to cover the functionality of the Auth CRD
+    [Tags]      Smoke
+    ...         Operator
+    ...         RHOAIENG-18846
+    ...         ExcludeOnODH
+    ${rc}       ${odhdashboardconfig_groups}=    Run And Return Rc And Output
+    ...    oc get odhdashboardconfig odh-dashboard-config -n redhat-ods-applications -o jsonpath='{.spec.groupsConfig}'
+    Should Be Equal As Integers     ${rc}       0
+    ${rc}       ${auth_cr_groups}=      Run And Return Rc And Output
+    ...                     oc get auth auth -o jsonpath='{.spec}'
+    Should Be Equal As Integers     ${rc}       0
+    ${auth_cr_groups_str}=      Remove String       ${auth_cr_groups}   [   ]
+    IF    """${odhdashboardconfig_groups}""" != """${auth_cr_groups_str}"""
+        Fail        User Groups on Auth Cr and OdhDashboardConfig DO NOT match
+    ELSE
+        Log To Console      User Groups on Auth Cr and OdhDashboardConfig match
+    END
+    Get User Groups From Auth Cr And Check Rolebinding Exists       adminGroups     admingroup-rolebinding
+    Get User Groups From Auth Cr And Check Rolebinding Exists       allowedGroups   allowedgroup-rolebinding
+
+
+*** Keywords ***
+Auth Crd Suite Setup
+    [Documentation]    Suite setup
+    RHOSi Setup
+
+Auth Crd Suite Teardown
+    [Documentation]    Suite teardown
+    RHOSi Teardown
+
+Get User Groups From Auth Cr And Check Rolebinding Exists
+    [Documentation]    Get User Groups From Auth CR And Check Rolebinding Exists
+    [Arguments]    ${group}    ${rolebinding}
+    ${rc}    ${out}=    Run And Return Rc And Output        oc get auth auth -o jsonpath='{.spec.${group}}'
+    Should Be Equal As Integers    ${rc}    0
+    ${groups_str}=    Remove String    ${out}    [    ]
+    @{groups_list}=    Split String    ${groups_str}    ,
+    FOR    ${user}    IN    @{groups_list}
+        Log To Console    ${user}
+        ${rc}=    Run And Return Rc
+        ...    oc get RoleBinding ${rolebinding} -n redhat-ods-applications -o jsonpath='{.subjects[?(@.name==${user})]}'   #robocop: disable:line-too-long
+        Should Be Equal As Integers    ${rc}    0
```

Comment:
I think it will always return zero if the rolebinding with such name exists but jsonpath doesn't return anything. So I think we need to add a check for output being not empty.

Reply:
No, it fails, I added a spare user group to check negative 
![image](https://github.com/user-attachments/assets/6dae79d8-ff59-4b2b-9c86-34c37cb72e3f)

Reply:
Interesting, when I execute locally
```
$ oc get RoleBinding admingroup-rolebinding -n redhat-ods-applications -o jsonpath='{.subjects[?(@.name=="dedicated-admins")]}' 2>&1
$ echo "$?"
0
```

I get 0. Never mind.

Reply:
maybe you had dedicated-admins group added to the admin group

Reply:
oc get RoleBinding admingroup-rolebinding -n redhat-ods-applications -o jsonpath='{.subjects}'
returns only:
[{"apiGroup":"rbac.authorization.k8s.io","kind":"Group","name":"rhods-admins"}]
So no idea.